### ISSUE TITLE
Flickers from default TailwindCSS styling

### ISSUE THREAD

**Issue by Destaq** [saadeghi/daisyui#249]
The default TailwindCSS theme is applied for some time, even on a production build, when using DaisyUI.

I believe this is due to upgrading to Nuxt-Bridge and thus slightly changing the way in which TailwindCSS is imported.

You can see a minimum reproducible example [here](https://github.com/Destaq/daisyui-nuxt-bridge), by going to the examples/nuxtjs section.

https://user-images.githubusercontent.com/61620873/137582807-4ec2eec8-ab0a-437c-a15d-642447fda13a.mov

**Comment by Destaq** [comment]
@saadeghi Reopening this as I've found a significant issue when using `nuxt3`, even with the example you provided.

The problem is with supporting multiple themes. It only occurs when you have a `pages` directory, and thus Vue Router (as the [nuxt3 example](https://github.com/saadeghi/daisyui/tree/master/examples/nuxtjs-v3) for the repo does. If you add the following code to `pages/index.vue`:

```js
<script>
export default {
  setup() {
    useMeta({
      htmlAttrs: {
        "data-theme": "retro"
      }
    })
  },
};
</script>
```

then there will be a flicker from the default DaisyUI styling (either dark or light mode, depending on the system), to the chosen theme. The library is immediately used, but it utilizes one of these two things before flickering to the right theme.

Of course, this can be changed by manually setting the `data-theme` at the top in the simplified example I provided, but would not solve the issue, as the `data-theme` in the `<script>` will be dynamic, based on [loading cookies](https://v3.nuxtjs.org/docs/usage/cookies/). An example of this is below.

```js
<script>
import { useCookie } from "#app";

export default {
  setup() {
    const theme = useCookie("theme") || "garden";

    useMeta({
      htmlAttrs: {
        "data-theme": theme,
      },
      title: "My Nuxt App",
    });

    return { theme };
  },
};
</script>
```

Any tips to get this working with no flicker?

**Comment by Destaq** [comment]
This only occurs when Server-Side-Rendering is set to true (the default). The following `nuxt.config.js` configuration doesn't cause the issue.

```js
import { defineNuxtConfig } from 'nuxt3'

export default defineNuxtConfig({
  build: {
    postcss: {
      postcssOptions: {
        plugins: {
          tailwindcss: {},
          autoprefixer: {},
        }
      }
    },
  },
  ssr: false,
})
```

But any fix without changing the fundamental Nuxt app itself?

**Comment by saadeghi** [comment]
I did some tests.    
Instead of Tailwind file, I imported this little file:
```css
[data-theme="retro"]{
  background-color: black;
}
```
And when I refresh pages on Chrome and FireFox, I see flicker of a white page before it gets black. (Safari is fine)  
When I import this file:
```css
:root{
  background-color: black;
}
```
And I refresh the page, there's no flicker, page is black in every millisecond.  

It looks like Chrome and Firefox are reading attribute selector styles slower than other element selectors 🤔

**Comment by saadeghi** [comment]
Also this code will show yellow flicker before it turns black. but it should simply show black color with no delay.
```css
:root{
  --color: yellow;
}
[data-theme="retro"]{
  --color: black;
}
:root{
  background-color: var(--color);
}
```

I guess the problem is performance of attribute selectors on Chrome and Firefox.

**Comment by andreashaerter** [comment]
Just for-the-record: I am currently experiencing a comparable issue using

- tailwindcss: 3.0.23
- @tailwindcss/forms: 0.5.0
- @tailwindcss/typography: 0.5.2
- daisyui: 2.11.0

and Hugo (no JavaScript libraries or Frameworks involved; same effect when JS is disabled). Will try to produce an easy-to-follow setup to make it easier to reproduce the issue.

**Comment by andreashaerter** [comment]
A bit more information: I guess this issue is _not_ related to daisyUI. I am able to reproduce it on nearly every website using Tailwind as well as on a simple local site under following condition:

- **Chromium** based browser with  **NoScript-Extension** (this is essential)
- Site containing SVG images

Examples on tailwindui.com as well as a local Hugo experiment with daisyUI (cf. comment above):

![flicker-with-noscript-01](https://user-images.githubusercontent.com/228611/159189660-97ab9c92-53e7-49e7-809b-f203e3bc5c70.gif)

Reloading a website is more likely to trigger the effect than opening the website the first time. As [NoScript had trouble with Googles DeclarativeNetRequest API](https://groups.google.com/a/chromium.org/g/chromium-extensions/c/veJy9uAwS00/m/lz3XsiGeGgAJ) (like others), there might be some CSS magic which is simply getting to slow when these kind of extensions are involved.

However, I can reproduce some slightly comparable issues on other website but only Tailwind based are this recognizable due to the default color scheme. Even though it's an edge case: **Maybe one can find a way to make the default style flicker less recognizable?**

**Comment by saadeghi** [comment]
I agree this is not a problem from daisyUI because daisyUI is just a plugin that adds component class names to a Tailwind CSS build and at the end, it's up to the framework how to handle CSS (load it as a file or not/ load it before JS files, or inside JS or after JS render)

One way would be separating critical style (that is needed in the first view) and load them before first before any JS. But there's a big issue about that: CSS specificity. If it's a separate file or inline CSS it will mess with Tailwind's order of class names because it's important what class name comes first in a utility-first library.  

I think it's better to move this issue to discussions.
